### ISSUE TITLE
Consistent type to express date & time

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/semantic-conventions#660]
While looking at different semantic conventions I saw that different notations for time are used, i.e. UNIX timestamps as int and formatted date as strings:

https://github.com/open-telemetry/semantic-conventions/blob/a41f0e8961860205f37e585443d4de3d12d25c2a/docs/attributes-registry/messaging.md?plain=1#L117

https://github.com/open-telemetry/semantic-conventions/blob/a41f0e8961860205f37e585443d4de3d12d25c2a/docs/resource/cloud-provider/heroku.md?plain=1#L14

https://github.com/open-telemetry/semantic-conventions/blob/a41f0e8961860205f37e585443d4de3d12d25c2a/docs/messaging/rocketmq.md?plain=1#L25

https://github.com/open-telemetry/semantic-conventions/blob/a41f0e8961860205f37e585443d4de3d12d25c2a/docs/attributes-registry/tls.md?plain=1#L19

I think there should be one consistent way to express those OR at least a `_timestamp` should be equivalent to a unix timestamp and a `_datetime` should be a formatted string

**Comment by pyohannes** [comment]
Good point. ECS supports different representations for date & time, but at least has it clearly documented: https://www.elastic.co/guide/en/elasticsearch/reference/8.12/date.html

**Comment by svrnm** [comment]
via https://github.com/open-telemetry/semantic-conventions/pull/564#discussion_r1499543297 (@jsuereth)

> We had some in person discussions in the Specification meeting and the Semconv meeting.
> I'm not sure we have consensus here, but I do think we can likely make progress by sticking (consistently) to String-dates in ISO 8601 format in semconv (what you have).

I am wondering where this should be added, from what I see there is no file that describes attribute values in more details, beyond what is in https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common?

**Comment by lmolkova** [comment]
I think we addressed it:

https://github.com/open-telemetry/semantic-conventions/blob/cefd73a824531597a1ebd77e09c5eebe21665241/docs/how-to-write-conventions/README.md?plain=1#L98
